File: app/src/main/java/com/kelsos/mbrc/adapters/LibraryPagerAdapter.java
package com.kelsos.mbrc.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentStatePagerAdapter;
import com.kelsos.mbrc.R;
import com.kelsos.mbrc.annotations.Search;
import com.kelsos.mbrc.annotations.Search.Section;
import com.kelsos.mbrc.ui.fragments.BrowseAlbumFragment;
import com.kelsos.mbrc.ui.fragments.BrowseArtistFragment;
import com.kelsos.mbrc.ui.fragments.BrowseGenreFragment;
import com.kelsos.mbrc.ui.fragments.BrowseTrackFragment;

public class LibraryPagerAdapter extends FragmentStatePagerAdapter {
  private static final int count = 4;
  private final Context context;

  public LibraryPagerAdapter(FragmentActivity activity) {
    super(activity.getSupportFragmentManager());
    context = activity.getApplicationContext();
  }

  @Override
  public Fragment getItem(@Section int position) {
    switch (position) {
      case Search.SECTION_GENRE:
        return new BrowseGenreFragment();
      case Search.SECTION_ARTIST:
        return new BrowseArtistFragment();
      case Search.SECTION_ALBUM:
        return new BrowseAlbumFragment();
      case Search.SECTION_TRACK:
        return new BrowseTrackFragment();
      default:
        break;
    }
    return null;
  }

  @Override
  public int getCount() {
    return count;
  }

  @Override
  public CharSequence getPageTitle(@Section int position) {
    switch (position) {
      case Search.SECTION_ALBUM:
        return context.getString(R.string.label_albums);
      case Search.SECTION_ARTIST:
        return context.getString(R.string.label_artists);
      case Search.SECTION_GENRE:
        return context.getString(R.string.label_genres);
      case Search.SECTION_TRACK:
        return context.getString(R.string.label_tracks);
      default:
        return "";
    }
  }
}

File: app/src/main/java/com/kelsos/mbrc/adapters/LibraryPagerAdapter.kt
package com.kelsos.mbrc.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.kelsos.mbrc.annotations.Search
import com.kelsos.mbrc.ui.navigation.library.albums.BrowseAlbumFragment
import com.kelsos.mbrc.ui.navigation.library.artists.BrowseArtistFragment
import com.kelsos.mbrc.ui.navigation.library.gernes.BrowseGenreFragment
import com.kelsos.mbrc.ui.navigation.library.tracks.BrowseTrackFragment

class LibraryPagerAdapter(activity: FragmentActivity) : FragmentStateAdapter(activity) {

  override fun createFragment(position: Int): Fragment {
    return when (position) {
      Search.SECTION_GENRE -> BrowseGenreFragment()
      Search.SECTION_ARTIST -> BrowseArtistFragment()
      Search.SECTION_ALBUM -> BrowseAlbumFragment()
      Search.SECTION_TRACK -> BrowseTrackFragment()
      else -> throw IllegalStateException("invalid position")
    }
  }

  override fun getItemCount(): Int {
    return COUNT
  }

  companion object {
    const val COUNT = 4
  }
}
